src/coreclr/gc/gc.cpp

```diff
@@ -6966,6 +6972,11 @@ bool gc_heap::virtual_commit (void* address, size_t size, int bucket, int h_numb
                 exceeded_p = true;
             }
         }
+#ifdef COMMITTED_BYTES_SHADOW
+        if (!heap_hard_limit && !heap_hard_limit_oh[soh]) {
```

Comment:
do you need the check for heap_hard_limit_oh part? if not let's only check for the total hard limit